src/kernel/pgtbl.c

```diff
@@ -1,187 +1,234 @@
-	assert(!(orig_v & PGTBL_QUIESCENCE));
+	return chal_pgtbl_kmem_act(pt, addr, kern_addr, pte_ret);
+}
+/* End Function:pgtbl_kmem_act ***********************************************/
+
+/* Begin Function:tlb_quiescence_check ****************************************
+Description : Check if the TLB is quiescent. This function will return 1 if 
+              the quiescence time have passed since input timestamp. 0 if not.
+Input       : u64_t timestamp - The reference timestamp.
+Output      : None.
+Return      : int - If quiescent, 0; else 1.
+******************************************************************************/
+int
+tlb_quiescence_check(u64_t timestamp)
+{
+	return chal_tlb_quiescence_check(timestamp);
+}
+/* End Function:tlb_quiescence_check *****************************************/
+
+/* Begin Function:cap_memactivate *********************************************
+Description : Delegate a user page from one page table to another. This will be 
+              called by the capinv.c's system call path.
+Input       : struct captbl *ct - The capability table for the destination page table.
+              struct cap_pgtbl *pt - The source page table.
+              capid_t frame_cap - The virtual address of the cosframe to delegate.
+              vaddr_t vaddr - The virtual address of to delegate to on the destination page table.
+Output      : None.
+Return      : int  - If successful, 0; else an error code.
+******************************************************************************/
+int
+cap_memactivate(struct captbl *ct, struct cap_pgtbl *pt, capid_t frame_cap, capid_t dest_pt, vaddr_t vaddr)
+{
+	return chal_cap_memactivate(ct, pt, frame_cap, dest_pt, vaddr);
+}
+/* End Function:cap_memactivate **********************************************/
+
+/* Begin Function:pgtbl_activate **********************************************
+Description : Activate(create) a page table capability in the capability table
+              designated. At this point, the page table data structure page is 
+              already allocated.
+Input       : struct captbl *t - The master capability table.
+              unsigned long cap - The capability to the capability table that you want 
+                                  this newly created page table capability to be in.
+              unsigned long capin - The position that you want this page table capability
+                                    to be in.
+              pgtbl_t pgtbl - The page table data structure.
+              u32_t lvl - The level of this page table.
+Output      : None.
+Return      : int  - If successful, 0; else an error code.
+TODO        : 1. The captbl is named t instead of ct. Inconsistency.
+              2. cap and capin is badly named.
+              3. u32_t, fixed length types are not generic.
+******************************************************************************/
```

Comment:
Though those comments are beautiful, I have the same concern that they generate unnecessary redundancy.
For example
```pgtbl_t pgtbl``` is obviously a page table structure, not sure comments for it is necessary.
```struct captbl *t```: `t` is not a good name even we have a comment for it. (`ct` might be better, as you said)
So if we change `t` to `ct` in future, we need to change both code and comments for it. I think this will increase the possibility to have confusing code.
If you think those names are bad, you could directly change them. Adding comments cannot fundamentally solve the problem.
What is the different between Output and Return?
Don't we always have None Output in kernel functions?